editor/export/editor_export_platform_apple_embedded.cpp

```diff
@@ -629,9 +629,9 @@ void EditorExportPlatformAppleEmbedded::_fix_config_file(const Ref<EditorExportP
 			switch (image_scale_mode) {
 				case 0: {
 					String logo_path = get_project_setting(p_preset, "application/boot_splash/image");
-					bool is_on = get_project_setting(p_preset, "application/boot_splash/fullsize");
+					RenderingServer::SplashStretchMode stretch_mode = get_project_setting(p_preset, "application/boot_splash/stretch_mode");
 					// If custom logo is not specified, Godot does not scale default one, so we should do the same.
-					value = (is_on && logo_path.length() > 0) ? "scaleAspectFit" : "center";
+					value = (stretch_mode != RenderingServer::SplashStretchMode::SPLASH_STRETCH_MODE_DISABLED && logo_path.length() > 0) ? "scaleAspectFit" : "center";
```

Comment:
This should use other modes as well (`scaleAspectFit` - KEEP, `scaleAspectFill` - COVER, `scaleToFill` - EXPAND).